electrumx/server/block_processor.py

```diff
@@ -453,21 +462,29 @@ def unpac_staking_utxo_index(pk_script):
             for idx, txout in enumerate(tx.outputs):
                 # Ignore unspendable outputs
                 if is_unspendable(txout.pk_script):
-                    if len(txout.pk_script) == STAKING_HEADER_LENGHT:
+                    if len(txout.pk_script) in STAKING_HEADER_LENGTHS \
+                    and txout.pk_script[2:4] == b'\x53\x44':
+                        staking_idx = unpack_staking_utxo_index(txout.pk_script)
+
                         reversed_tx_hash = bytearray(tx_hash)
                         reversed_tx_hash.reverse()
-                        if self.daemon.request_staking(reversed_tx_hash.hex()) is None:
-                            continue
-                        staking_idx = unpac_staking_utxo_index(txout.pk_script)
-                    continue
+
+                        if staking_idx > len(tx.outputs) \
+                        or staking_idx < 0 \
```

Comment:
the condition must be `<=`. 0 index would point to the `OP_RETURN` script that is being parsed.